src/data/device_automation.ts

```diff
@@ -0,0 +1,39 @@
+import { HomeAssistant } from "../types";
+
+export interface DeviceTrigger {
+  platform: string;
+  device_id: string;
+  domain: string;
+  entity_id: string;
+  type?: string;
+  event?: string;
+}
+
+export interface DeviceTriggerList {
+  triggers: DeviceTrigger[];
+}
+
+export const fetchDeviceTriggers = (hass: HomeAssistant, deviceId: string) =>
+  hass.callWS<DeviceTriggerList>({
+    type: "device_automation/trigger/list",
+    device_id: deviceId,
+  });
+
+export const triggersEqual = (a: any, b: any) => {
+  if (typeof a !== typeof b) {
+    return false;
+  }
+
+  for (const property in a) {
```

Comment:
This is incorrect, now `triggersEqual({a: "yo"}, {a: "yo"}) === true`

Reply:
I would just compare each property that you care about. `tr1.device_id === tr2.device_id && …`

Reply:
Hmm, what's wrong with `triggersEqual({a: "yo"}, {a: "yo"}) === true`?

The reason for not just comparing known properties is that the frontend then needs to know about all trigger.

Reply:
I think that it makes sense that we know about all higher level properties in a trigger object. If we will allow options in the future, we can do that in an objects key.